src/codecs/pnm/decoder.rs

```diff
@@ -1262,3 +1266,17 @@ ENDHDR
         let _ = crate::load_from_memory(b"P391919 16999 1 1 9 919 16999 1 9999 999* 99999 N");
     }
 }
+
+#[test]
+fn read_samples_test() {
+    let data = vec![80, 54, 10, 52, 50, 57, 52, 56, 50, 57, 52, 56, 35, 56, 10, 52, 10, 48, 10, 12, 12, 56];
+    let mut decoder = PnmDecoder::new(&data[..]).unwrap();
+    assert_eq!(decoder.color_type(), ColorType::L8);
+    assert_eq!(decoder.dimensions(), (16, 16));
+    assert_eq!(decoder.subtype(), PnmSubtype::Graymap(SampleEncoding::Binary));
+    let mut image = vec![0; decoder.total_bytes() as usize];
+    decoder.read_image(&mut image).unwrap();
+    assert_eq!(image, vec![80, 54, 10, 52, 50, 57, 52, 56, 50, 57, 52, 56, 35, 56, 10, 52, 10, 48, 10, 12, 12, 56]);
+
```

Comment:
The asserts do not hold for that image. We're only interested in not crashing and having a valid eader. They could just be removed (please verify on your local machine that this works):

```suggestion
fn read_samples_test() {
    let data = vec![80, 54, 10, 52, 50, 57, 52, 56, 50, 57, 52, 56, 35, 56, 10, 52, 10, 48, 10, 12, 12, 56];
    let mut decoder = PnmDecoder::new(&data[..]).unwrap();
    let mut image = vec![0; decoder.total_bytes() as usize];
    let _ = decoder.read_image(&mut image);
}
```